internal/server/config/config_test.go

```diff
@@ -0,0 +1,177 @@
+package config
+
+import (
+	"flag"
+	"os"
+	"testing"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
+
+	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/urfave/cli"
+)
+
+func TestNew(t *testing.T) {
+	tests := []struct {
+		name string
+		args []string
+		want Config
+	}{
+		{
+			name: "happy path",
+			args: []string{"-quiet", "--no-progress", "--reset", "--skip-update"},
+			want: Config{
+				Quiet:      true,
+				NoProgress: true,
+				Reset:      true,
+				SkipUpdate: true,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			app := &cli.App{}
+			set := flag.NewFlagSet("test", 0)
+			set.Bool("quiet", false, "")
+			set.Bool("no-progress", false, "")
+			set.Bool("reset", false, "")
+			set.Bool("skip-update", false, "")
+
+			c := cli.NewContext(app, set, nil)
+			_ = set.Parse(tt.args)
+
+			tt.want.context = c
```

Comment:
What's actually asserted in theses tests?

Reply:
`tt.want` is a `Config` type which has many fields, but `context` is a pointer. So, I wanted to do `assert.Equal(t, tt.want, c, tt.name)` except for `tt.want.context` and `tt.want.logger`. I had no idea to do it, so I copied the value (ctx) to the wanted value (tt.want.context). 
Do you have any idea to compare an actual value and an expected value if they contain a pointer?

Reply:
I wish we had such utility in Golang which I used a lot in my Java days -> https://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#field-by-field-ignoring

We can improve it later on though. Just remember to add assert which refers to `tt.want`. I don't see it, so even if you set the `context` we don't actually do asserts.

Reply:
Yes, that is exactly what I want. I didn't find it in `testify/assert`.

That was my fault. I added the assertion.
https://github.com/aquasecurity/trivy/pull/303/commits/303bf698c13b7acd83d43d9316effea680230395

Reply:
Yes, that is exactly what I want. I didn't find it in testify/assert.

That was my fault. I added the assertion.
https://github.com/aquasecurity/trivy/pull/303/commits/303bf698c13b7acd83d43d9316effea680230395